compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
+  SmallVector<std::string> entryPointNames;
+  for (auto exportOp : variantOp.getExportOps()) {
+    entryPointNames.emplace_back(exportOp.getSymName());
   }
 
-  // Create a temporary work directory needed for subsequent commands.
-  FailureOr<std::string> workDir =
-      Artifact::createTemporaryDirectory(variantOp.getName());
-  if (failed(workDir)) {
+  if (entryPointNames.size() == 0) {
+    return moduleOp.emitOpError("Entry point missing");
+  } else if (entryPointNames.size() > 1) {
+    return moduleOp.emitOpError("Multiple entry points NYI");
+  }
+
+  SmallVector<StringRef> CmdArgs{aie2xclbin,
+                                 inputMlirPath,
+                                 "--peano",
+                                 options.peanoInstallDir,
+                                 "--xclbin-name",
+                                 XCLBinPath,
+                                 "--ipu-insts-name",
+                                 IPUInstPath,
+                                 "--xclbin-kernel-name",
+                                 entryPointNames[0],
+                                 "--tmpdir",
+                                 workDir};
+  // Update the linker search path to find libcdo_driver.so
+  SmallString<128> LibPath(options.vitisInstallDir);
+  llvm::sys::path::append(LibPath, "aietools", "lib", "lnx64.o");
+  const char *originalLDPath = ::getenv("LD_LIBRARY_PATH");
+  std::string newLDPath;
+  if (originalLDPath == nullptr) {
+    newLDPath = LibPath.str();
+  } else {
+    newLDPath =
+        llvm::join_items(llvm::sys::EnvPathSeparator, LibPath, originalLDPath);
+  }
+  newLDPath = "LD_LIBRARY_PATH=" + newLDPath;
+  std::string newVitis = "VITIS=" + options.vitisInstallDir;
+  SmallVector<StringRef> CmdEnv{newLDPath, newVitis};
+  const char *originalPath = ::getenv("PATH");
+  std::string newPath;
+  if (originalPath != nullptr) {
+    newPath = originalPath;
+    // Amend the dll search path
+#ifdef _WIN32
+    newPath = llvm::join_items(llvm::sys::EnvPathSeparator, LibPath, newPath);
+#endif
+    newPath = "PATH=" + newPath;
+    CmdEnv.push_back(newPath);
+  }
+  int result = llvm::sys::ExecuteAndWait(CmdArgs[0], CmdArgs, CmdEnv);
+  if (result != 0) {
     return moduleOp.emitOpError(
-        "failed to create temporary working directory for xclbin generation");
+        "Failed to produce an XCLBIN with external tool");
   }
-  llvm::outs() << "Temporary WorkDir : " << workDir.value() << "\n";
 
-  // Generate the core elf file.
-  {
-    OpBuilder::InsertionGuard g(executableBuilder);
-    executableBuilder.setInsertionPoint(moduleOp);
-    auto clonedModuleOp =
-        dyn_cast<ModuleOp>(executableBuilder.clone(*moduleOp.getOperation()));
-    if (failed(generateCoreElfFiles(moduleOp, objFile.value(), workDir.value(),
-                                    getOptions(), toolkit))) {
-      return failure();
+  std::vector<uint32_t> ipuInstrs;
+
+  std::ifstream instr_file(static_cast<std::string>(IPUInstPath));
```

Comment:
I am not sure why we are doing this? I would assume that we could just open it, read it as a file and just embed it without having to do any processing here.

Reply:
The function I call in the tool writes hexadecimal.

Reply:
I thought even the binary IREE writes to vmfb in hex. @nirvedhmeshram can you comment on this.

Reply:
I think the file is a string of hex values, and we are converting it to a vector of uint32_t for the vmfb here. So it seems required, unless the tool can allow to keep the vector in memory and we never go to disk.